controlplane/src/core/repositories/FederatedGraphRepository.ts

```diff
@@ -251,8 +251,42 @@ export class FederatedGraphRepository {
         await targetRepo.updateReadmeOfTarget({ id: data.targetId, readme: data.readme });
       }
 
+      const haveLabelMatchersChanged = (() => {
```

Comment:
Please put this in a helper function on the class or somewhere else so it is testable.

Reply:
Done